pkg/resources/helpers.go

```diff
@@ -330,3 +330,183 @@ func ListDiff[T comparable](beforeList []T, afterList []T) (added []T, removed [
+				kmsKeyIdEqual := s1.GCSStorageLocationParams.Encryption.KmsKeyId == nil && s2.GCSStorageLocationParams.Encryption.KmsKeyId == nil ||
+					(s1.GCSStorageLocationParams.Encryption.KmsKeyId != nil && s2.GCSStorageLocationParams.Encryption.KmsKeyId != nil &&
+						*s1.GCSStorageLocationParams.Encryption.KmsKeyId == *s2.GCSStorageLocationParams.Encryption.KmsKeyId)
+				return typeEqual && kmsKeyIdEqual, nil
+			}
+		}
+	case sdk.StorageProviderAzure:
+		return s1.AzureStorageLocationParams.Name == s2.AzureStorageLocationParams.Name &&
+			s1.AzureStorageLocationParams.AzureTenantId == s2.AzureStorageLocationParams.AzureTenantId &&
+			s1.AzureStorageLocationParams.StorageBaseUrl == s2.AzureStorageLocationParams.StorageBaseUrl, nil
+	}
+
+	return false, nil
+}
+
+// Returns a copy of the given storage location with a 'temp_' prefix on the Name field
+func CopyStorageLocationWithTempName(
+	storageLocation sdk.ExternalVolumeStorageLocation,
+) (sdk.ExternalVolumeStorageLocation, error) {
+	storageProvider, err := GetStorageLocationStorageProvider(storageLocation)
+	if err != nil {
+		return sdk.ExternalVolumeStorageLocation{}, err
+	}
+
+	currName, err := GetStorageLocationName(storageLocation)
+	if err != nil {
+		return sdk.ExternalVolumeStorageLocation{}, err
+	}
+
+	newName := fmt.Sprintf("temp_%s", currName)
+	var tempNameStorageLocation sdk.ExternalVolumeStorageLocation
+	switch storageProvider {
+	case sdk.StorageProviderS3, sdk.StorageProviderS3GOV:
+		tempNameStorageLocation = sdk.ExternalVolumeStorageLocation{
+			S3StorageLocationParams: &sdk.S3StorageLocationParams{
+				Name:                 newName,
+				StorageProvider:      storageLocation.S3StorageLocationParams.StorageProvider,
+				StorageBaseUrl:       storageLocation.S3StorageLocationParams.StorageBaseUrl,
+				StorageAwsRoleArn:    storageLocation.S3StorageLocationParams.StorageAwsRoleArn,
+				StorageAwsExternalId: storageLocation.S3StorageLocationParams.StorageAwsExternalId,
+				Encryption:           storageLocation.S3StorageLocationParams.Encryption,
+			},
+		}
+	case sdk.StorageProviderGCS:
+		tempNameStorageLocation = sdk.ExternalVolumeStorageLocation{
+			GCSStorageLocationParams: &sdk.GCSStorageLocationParams{
+				Name:           newName,
+				StorageBaseUrl: storageLocation.GCSStorageLocationParams.StorageBaseUrl,
+				Encryption:     storageLocation.GCSStorageLocationParams.Encryption,
+			},
+		}
+	case sdk.StorageProviderAzure:
+		tempNameStorageLocation = sdk.ExternalVolumeStorageLocation{
+			AzureStorageLocationParams: &sdk.AzureStorageLocationParams{
+				Name:           newName,
+				StorageBaseUrl: storageLocation.AzureStorageLocationParams.StorageBaseUrl,
+				AzureTenantId:  storageLocation.AzureStorageLocationParams.AzureTenantId,
+			},
+		}
+	}
+
+	return tempNameStorageLocation, nil
+}
+
+func GetStorageLocationName(s sdk.ExternalVolumeStorageLocation) (string, error) {
```

Comment:
These functions are should be `sdk. ExternalVolumeStorageLocation` methods.

Reply:
Happy to move them, which of the generated files would you put custom functions in normally?

Reply:
You can simply add it in `external_volumes_gen.go`.